include/gadget/types.h

```diff
@@ -3,6 +3,12 @@
 #ifndef __TYPES_H
 #define __TYPES_H
 
+// Networking gadgets don't include vmlinux.h, so we need to define few types
+// needed by other headers.
+#ifdef GADGET_TYPE_NETWORKING
+#include <stdbool.h>
+#endif
```

Comment:
Why do you need `stdbool.h`?

In order to fix #2371, I would like to be possible for gadget authors to write networking gadgets that include `vmlinux.h`. So I prefer not to rely on `#ifdef GADGET_TYPE_NETWORKING` to detect the presence of `vmlinux.h`. Instead, I suggest to use the same code pattern as in `sockets-map.h`:

```c
// Types such as __u64 below requires to include either <linux/types.h> or
// <vmlinux.h> before. We can't include both because they are incompatible. Let
// the gadget choose which one to include.
#if !defined(__VMLINUX_H__) && !defined(_LINUX_TYPES_H)
#error "Include <linux/types.h> or <vmlinux.h> before including this file."
#endif
```

Reply:
> Why do you need stdbool.h?

To get a definition of bool that we're using in a lot of places. 

I changed the approach, is it fine now?

Reply:
Your change looks good to me.